plugin/firewall/README.md

```diff
@@ -0,0 +1,112 @@
+# firewall
+
+## Name
+
+*firewall* - enables filtering on query and response using direct expression as policy.
+
+## Description
+
+The firewall plugin defines a rule list of expressions that triggers workflow action on the DNS query or its response.
+A rule list is an ordered set of rules that are evaluated in sequence.
+Each rule has two parts: an action and an expression. When the rule is evaluated,
+first the expression is evaluated.
+- If the expression evaluates to `true` the action is performed on the query and the rule list evaluation ceases.
+- If the expression does not evaluates to `true` then next rule in sequence is evaluated.
+
+
+## Syntax
+
+~~~ txt
+firewall DIRECTION {
+    ACTION EXPRESSION
+    ACTION EXPRESSION
+    ...
+}
+~~~~
+
+* **DIRECTION** indicates if the _rule list_ will be applied to queries or responses. It can be `query` or `response`.
+
+* **ACTION** defines the workflow action to apply to the DNS operation if the **EXPRESSION** evaluates to the boolean `true`
+action is one of:
+  - `allow` : continue the DNS resolution process
+  - `refuse` : interrupt the DNS resolution, just return the REFUSE result
```

Comment:
what's "result"? Do you mean you return a packet with rcode (response code) REFUSE?

Reply:
yes. will change the text with "reply with XXXX code".